### PR TITLE
fix: restrict upcoming events on basis of ticket availability

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5110

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/events.py [hunk 1]
```diff
@@ -850,6 +850,7 @@ def query(self, view_kwargs):
                 Event.ends_at > current_time,
                 Event.state == 'published',
                 Event.privacy == 'public',
+                Event.tickets.any(and_(Ticket.deleted_at == None, Ticket.price > 0, Ticket.is_hidden == False)),
```
Comment (iamareebjamal) [inline app/api/events.py:853]: Promoted events should be shown anyway

Comment (iamareebjamal) [reply app/api/events.py:853]: Also price was an example. I don't think price is a requirement

Comment (maze-runnar) [reply app/api/events.py]: > Also price was an example. I don't think price is a requirement

ok

File: app/api/events.py [hunk 2]
```diff
@@ -859,6 +858,7 @@ def query(self, view_kwargs):
                         Event.event_type_id != None,
                         Event.event_topic_id != None,
                         Event.event_sub_topic_id != None,
+                        Event.tickets.any(and_(Ticket.deleted_at == None, Ticket.price > 0, Ticket.is_hidden == False)),
```
Comment (iamareebjamal) [inline app/api/events.py]: Read my previous comment about price

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**